.github/workflows/verify.yaml

```diff
@@ -32,6 +32,34 @@ jobs:
           reporter: 'github-check'
           filter_mode: 'nofilter'
 
+  Dependency-analysis:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'gradle'
+
+      - name: Dependency rules report # possible severity values: <'fail'|'warn'|'ignore'>
+        run: ./gradlew buildHealth applyDependencyRules -Pdependency.analysis=warn -Pdependency.analysis.clear.artifacts=false
+
+      - name: Dependency analysis report
+        run: cat build/reports/dependency-analysis/build-health-report.txt
+
+  Unit-Test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
+      - name: Setup Gradle cache
```

Comment:
What's the difference of using `gradle-build-action` instead of the `cache: 'gradle'` attribute on the `setup-java`?
If this is better could be used in every job/workflow?

Reply:
The gradle build action has [some benefits](https://github.com/gradle/gradle-build-action#why-use-the-gradle-build-action) over the `setup-java` approach. Additionally `gradle.properties` are not included in the cache checksum when using `setup-java`. But it is true that it is unrelated to this PR, and could be done in a separate cleanup PR. I can revert this here and open a separate PR 👍🏻